src/test/kotlin/no/nav/helse/spion/web/integration/ArbeidsgiverApiTests.kt

```diff
@@ -0,0 +1,52 @@
+package no.nav.helse.spion.web.integration
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.request.receive
+import io.ktor.server.testing.handleRequest
+import io.ktor.server.testing.withTestApplication
+import io.ktor.util.KtorExperimentalAPI
+import no.nav.helse.spion.auth.MockAuthRepo
+import no.nav.helse.spion.domene.AltinnOrganisasjon
+import no.nav.helse.spion.web.spionModule
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Test
+import org.koin.core.get
+import org.koin.ktor.ext.get
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+@KtorExperimentalAPI
+class ArbeidsgiverApiTests : ControllerIntegrationTestBase() {
```

Comment:
Burde vi ha noen negative tester også? Som beviser at vi støtter feilhåndtering.

Reply:
Generelt er det smart å teste både false negatives og false positives. Altså at ikke bare virker det når det skal, men også at det ikke virker når det ikke skal.

Reply:
Jepp, det burde vi, men jeg vet ikke hvordan vi skal håndtere det, og har på en måte "scopet" det ut. Må lære mer om Ktor og hvordan det kan håndteres, samt at vi må kanskje snakke om det?

Det jeg er mest vant til er å bruke rammeverket sin støtte for å konvertere exception typer til HTTP-koder, og koverterer en payload til JSON med en callId tilbake til klienten for feilsøking i loggene:

HTTP 500
{
exception: 'no.nav.JuiceException',
message: 'Du er tom for juice',
callId: GUID-FOR-LOGGSØK
}